test/master-playlist-controller.test.js

```diff
@@ -323,6 +323,56 @@ QUnit.test('fast quality change resyncs audio segment loader', function(assert)
   assert.equal(resets, 0, 'does not reset the audio segment loader when media changes');
 });
 
+QUnit.test('resets everything for a fast quality change', function(assert) {
+  let resyncs = 0;
+  let resets = 0;
+  let removeFuncArgs = {};
+
+  this.masterPlaylistController.mediaSource.trigger('sourceopen');
+  // master
+  this.standardXHRResponse(this.requests.shift());
+  // media
+  this.standardXHRResponse(this.requests.shift());
+
+  let segmentLoader = this.masterPlaylistController.mainSegmentLoader_;
+
+  segmentLoader.resyncLoader = function() {
```

Comment:
minor: but for this (and below) we can save some lines using arrows, e.g., `() => resyncs++`